mitreattack/navlayers/README.md

```diff
@@ -249,3 +254,64 @@ conf.load_from_file(filename="path/to/poster/config.json")
+The module expects users to specify the domain and source during the initialization, and the type of entity whenever calling the generation method. 
+
+### OverviewGenerator()
+```python
+x = OverviewGenerator(source='taxii', domain='enterprise')
+```
+As stated above, the initialization for the generator expects the user to provide the domain and source. Valid options for domain include `enterprise`, `mobile`, and `ics`. 
+Valid options for source include `taxii`, and `local`, the latter of which requires the optional `local` parameter with a path to the local data store.
+
+### generate_layer()
+```python
+output_layer = x.generate_layer(obj_type='group')
+```
+This method creates and returns a ATT&CK Layer object (as defined within mitreattack-python core), with the appropriate notations for the given parameters. Valid `obj_type` values include `group`, `software`, and `mititagtion`.
+#### Example Usage
+```python
+from mitreattack.navlayers import OverviewGenerator
+
+x = OverviewGenerator(source='taxii', domain='mobile')
+output_layer_software = x.generate_layer(obj_type='software')
+output_layer_mitigation = x.generate_layer(obj_type='mitigation')
+
+output_layer_software.to_file(filename='demo_software_layer.json')
+print(output_layer_mitigation.to_str())
+```
+
+## usage_generator.py
+usage_generator provides the UsageGenerator class, which allows users to create either an Enterprise, Mobile, or ICS matrix containing notations on techniques corresponding to a specific Group, Software, or Mitigation. 
+The form of this notation is that each utilized technique will have a score of 1, and where the comment provided is the description of the entity. 
+The module expects users to specify the domain and source during the initialization, and to provide the specific entity when calling the generation method.
+
+### UsageGenerator()
+```python
+y = UsageGenerator(source='taxii', domain='enterprise')
+```
+As stated above, the initialization for the generator expects the user to provide the domain and source. Valid options for domain include `enterprise`, `mobile`, and `ics`. 
+Valid options for source include `taxii`, and `local`, the latter of which requires the optional `local` parameter with a path to the local data store.
+
+### generate_layer()
+```python
+y.generate_layer(match='APT29')
+```
+This method creates and returns a ATT&CK Layer object (as defined within mitreattack-python core), with the appropriate notations for the given parameters. Valid `match` values can be either a name, an alias/associated name, or ATT&CK ID.
+
+#### Example Usage
+```python
+from mitreattack.navlayers import UsageGenerator
+
+y = UsageGenerator(source='taxii', domain='enterprise')
+output_layer_apt1 = y.generate_layer(match='Comment Panda')
+output_layer_androrat = y.generate_layer(match='S0292')
+output_layer_data_backup = y.generate_layer(match='Data Backup')
```

Comment:
I'm thrilled that the "match" API is this easy to use. Great job designing it! I think the community is going to get a lot out of this functionality.

Reply:
I agree, the way it works in the current implementation is crazy simple and straightforward, at least from the user's perspective.